manager/client/client.go

```diff
@@ -1190,6 +1191,99 @@ WaitForAllData:
 	return aprs, nil
 }
 
+// IndexingStatus the head of the chain and show the difference with indexed block
+func (hc *Client) IndexingStatus(ctx context.Context, nv NetworkVersion) (info shared.IndexingStatusInfo, err error) {
+	defer hc.recoverPanic()
+
+	timer := metrics.NewTimer(callDurationIndexingStatus)
+	defer timer.ObserveDuration()
+
+	blockWithMeta := shared.BlockWithMeta{
+		Network: nv.Network,
+		Version: nv.Version,
+		ChainID: nv.ChainID,
+	}
+
+	lastBlockOfChain := &shared.Block{}
+
+	reqs := []structs.TaskRequest{}
+	req, err := hc.createTaskRequest(ctx, nv, "", 0, shared.ReqIDChainHeadBlock)
+	if err != nil {
+		return info, err
+	}
+	reqs = append(reqs, req)
+
+	hc.logger.Info("[Client] Sending request data:", zap.Any("requests", reqs))
+	var respAwait *structs.Await
+	if len(reqs) > 0 {
+		respAwait, err = hc.sender.Send(reqs)
+		if err != nil {
+			hc.logger.Error("[Client] Error sending data", zap.Error(err))
+			return info, fmt.Errorf("error sending data in IndexingStatus: %w", err)
+		}
+
+		defer respAwait.Close()
+	}
+
+	buff := &bytes.Buffer{}
+	dec := json.NewDecoder(buff)
+
+	var receivedFinals int
+WaitForAllData:
+	for {
+		if len(reqs) == 0 {
+			break
+		}
+
+		select {
+		case <-ctx.Done():
+			return info, errors.New("request timed out")
+		case response, ok := <-respAwait.Resp:
+			if !ok {
+				return info, fmt.Errorf("resp channel closed")
+			}
+			hc.logger.Debug("[Client] Get chain head received data:", zap.Any("type", response.Type), zap.Any("response", response))
+
+			if response.Error.Msg != "" {
+				return info, fmt.Errorf("error getting response: %s", response.Error.Msg)
+			}
+
+			if response.Type == "ChainHeadBlock" {
+				buff.Reset()
+				buff.ReadFrom(bytes.NewReader(response.Payload))
+				err = dec.Decode(lastBlockOfChain)
+				if err != nil {
+					return info, fmt.Errorf("error decoding latest block: %w", err)
+				}
+			}
+
+			if response.Final {
+				receivedFinals++
+			}
+
+			if receivedFinals == len(reqs) {
+				hc.logger.Info("[Client] Received IndexingStatus", zap.Any("network", nv.Network), zap.Any("chain_id", nv.ChainID), zap.Int("latest_block_of_chain", 1))
+				break WaitForAllData
+			}
+		}
+	}
```

Comment:
There's only one request, so you don't need a WaitForAllData loop. There will only be one response on respAwait.Resp, so you can simplify this (don't need to track finals, or use a decoder buffer, etc):

something like this should work fine
```suggestion
	for {
		select {
		case <-ctx.Done():
			return info, errors.New("request timed out")
		case response, ok := <-respAwait.Resp:
			if !ok {
				return info, fmt.Errorf("resp channel closed")
			}
			hc.logger.Debug("[Client] Get chain head received data:", zap.Any("type", response.Type), zap.Any("response", response))

			if response.Error.Msg != "" {
				return info, fmt.Errorf("error getting response: %s", response.Error.Msg)
			}

			if response.Type != "ChainHeadBlock" {
			    return info, fmt.Errorf("error getting ChainHeadBlock: wrong response type: %s", response.Type)
			}
			
			if err = json.Unmarshal(response.Payload, lastBlockOfChain); err != nil {
			    return info,  fmt.Errorf("error unmarshalling latest block: %s", response.Error.Msg)
			}
		}
	}
```

Reply:
👍